lib/messages.ts

```diff
@@ -0,0 +1,84 @@
+import { expect, Page } from '@playwright/test';
+
+export enum Limit {
+  ten = 10,
+  twenty = 20,
+  fifty = 50
+}
+
+export enum FilterGroup {
+  offset = 'Offset',
+  timestamp = 'Timestamp',
+  epoch = 'Epoch timestamp',
+  latest = 'Latest messages'
+}
+
+export const setPartition = async function (page: Page, partition: string) {
+  await page.locator('[aria-label="Specify partition value"]').click();
+  await page.locator('[aria-label="Specify partition value"]').fill(partition);
+};
+
+export const pickFilterOption = async function (page: Page, option: FilterGroup) {
+  await page.locator('[data-testid="filter-group"]').click();
+  switch (option) {
+    case FilterGroup.offset: {
+      await page.locator('a:text-is("' + FilterGroup.offset + '")').click();
+      break;
+    }
+    case FilterGroup.timestamp: {
+      await page.locator('a:text-is("' + FilterGroup.timestamp + '")').click();
+      break;
+    }
+    case FilterGroup.epoch: {
+      await page.locator('a:text-is("' + FilterGroup.epoch + '")').click();
+      break;
+    }
+    case FilterGroup.latest: {
+      await page.locator('a:text-is("' + FilterGroup.latest + '")').click();
+      break;
+    }
+  }
+};
+
+export const setOffeset = async function (page: Page, offset: string) {
```

Comment:
the following 4 methods and `setPartition` can be refactored in one:

```ts
export const setField = async function (page: Page, locator: string, value: string) {
  await page.locator(locator).click();
  await page.locator(locator).fill(value);
};
```

Reply:
I would say just 3, limit is a completely different component below this grouped filter and I would say the current approach is better in terms of maintainability and readability. 
I don't like the idea of every time in the code you would be passing a locator of what you want to set up. We want to avoid having to replace one locator across multiple places should something in the UI change.

Reply:
`setLimit` is different that's correct.

if you REALLY want to keep the locators in the library the comment still applies:
- implement `setField` as a non-exported method
- use it to refactor the rest

Reply:
Good idea, done